src/AWS.Deploy.CLI/Extensions/CustomServiceCollectionExtension.cs

```diff
@@ -0,0 +1,61 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using AWS.Deploy.CLI.Commands;
+using AWS.Deploy.CLI.Commands.TypeHints;
+using AWS.Deploy.CLI.Utilities;
+using AWS.Deploy.Common;
+using AWS.Deploy.Common.Extensions;
+using AWS.Deploy.Common.IO;
+using AWS.Deploy.Orchestration;
+using AWS.Deploy.Orchestration.CDK;
+using AWS.Deploy.Orchestration.Data;
+using AWS.Deploy.Orchestration.Utilities;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace AWS.Deploy.CLI.Extensions
+{
+    public static class CustomServiceCollectionExtension
+    {
+        /// <summary>
+        /// Extension method for <see cref="IServiceCollection"/> that injects essential app dependencies.
+        /// It is safer to use singleton instances for dependencies because every command (ex. deploy, list-deployments) run as a separate instance.
+        /// </summary>
+        /// <param name="serviceCollection"><see cref="IServiceCollection"/> instance that holds the app dependencies.</param>
+        public static void AddCustomServices(this IServiceCollection serviceCollection)
```

Comment:
nit: Change to `AddDeployToolCli()`? 

Down the line Visual Studio implementation might have a similar method like `AddDeployToolVisaulStudio()`